File: angular1/module/app/MessageTypes/MessageMethodResult.ts
module goTransport {

    export class MessageMethodResult extends Message{
        static type = MessageType.MessageTypeMethodResult;

        constructor(private result: boolean = false, private parameters: Array<any> = null) {
            super(MessageMethod.type);
        }

        validate(): Error {
            console.log('validating method result', this.reply);
            return null;
        }

        run(): Error {
            console.log('Running method result', this.reply);
            return null;
        }

    }

    MessageDefinition.set(MessageMethod.type, MessageMethod);
}

File: transport/lib/messageType/MessageMethodResult.go
package messageType

import (
	"github.com/iain17/goTransport/transport/lib/Message"
)

type messageMethodResult struct {
	Message.Message
	Result       bool `json:"result"`
	Parameters []interface{} `json:"parameters"`
}

func init() {
	Message.Set(NewMessageMethodResult(false, nil))
}

func NewMessageMethodResult(result bool, parameters []interface{}) *messageMethodResult {
	return &messageMethodResult{
		Message: Message.NewMessage(Message.MessageTypeMethodResult),
		Result: result,
		Parameters: parameters,
	}
}

func (message messageMethodResult) Sending() error {
	return nil
}

func (message messageMethodResult) Received() error {
	return 	nil
}